certificate_agent.py

```diff
@@ -97,16 +97,16 @@ def main():
             designation = xqueue_body.get('designation', None)
             if designation:
                 designation = designation.encode('utf-8')
-            if last_course != course_id:
+            if not CertificateGen.reuse_cert(cert, course_id, designation):
```

Comment:
maybe since we're refactoring this conditional let's move the definitions from above down and combine them:

```python
if CertificateGen.reuse_cert(cert, course_id, designation):
    cert = None
else:
    ...
```